x-pack/plugins/security_solution/public/resolver/test_utilities/simulator/index.tsx

```diff
@@ -166,6 +191,14 @@ export class Simulator {
     }
   }
 
+  /**
+   * Find the graph container by the database document ID.
+   * This better allows us to test when new data is loaded into an existing resolver
+   */
+  public processResolverGraph(databaseDocumentID: string): ReactWrapper {
```

Comment:
Another possible approach. Create a mock data access layer that returns different data based on the databaseDocumentID. Load resolver, then change the databaseDocumentID props. New data will be loaded. You will be able to verify that based on the data in the panel and the graph. maybe?

Reply:
I like this idea, but the more I thought about it, I realized I would be testing functionality that we don't currently have in the UI, as far as I can tell, so figured it would be best to leave this till we figure out what that implementation may look like. 🤷‍♂️